pkg/ingress/config_types.go

```diff
@@ -238,7 +240,7 @@ type HTTPRequestMethodConditionConfig struct {
 
 func (c *HTTPRequestMethodConditionConfig) validate() error {
 	if len(c.Values) == 0 {
-		return errors.New("values cannot be empty")
+		return errors.Wrap(errConditionValidation, "values cannot be empty")
```

Comment:
Prefer fmt.Errorf() and `%w` for wrapping errors.

But this isn't the use case for wrapping; one should probably use a custom error type and a constructor.

Reply:
Agree regarding to `fmt.Errorf()`.

> But this isn't the use case for wrapping

What' are cases for wrapping and what for creation custom type? It will be good to understand for me to make it properly in the future. Thanks!

Reply:
The canonical case for wrapping is when you received an error from doing a call and you want to add information about the context in which the error occurred. I tend to start the wrapping message with a gerund, for example "writing file %q".

A custom error type is for when the caller needs to programmatically check if a particular error condition occurred. In this case, the caller wants to know if there was an invalid rule condition.

Reply:
@johngmyers Thanks a lot!